pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -398,6 +411,166 @@ func (c *Controller) setupIndex() *AmbientIndex {
 			}
 		},
 	}
+
+	idx.servicesMap = make(map[types.NamespacedName]*model.Service)
+	idx.handleServiceEntry = func(svc *model.Service, event model.Event) {
+		idx.mu.Lock()
+		defer idx.mu.Unlock()
```

Comment:
nit; move this locl below the nil check on line 420

Reply:
addressed in https://github.com/istio/istio/pull/45621/commits/100522bd1b4d1e25907c0f32943f3bb87cf75f1e